modules/app_lb/main.tf

```diff
@@ -96,11 +96,21 @@ resource "azurerm_application_gateway" "default" {
     priority                   = 1
   }
 
+  dynamic "ssl_certificate" {
+    for_each = var.byoc ? [1] : []
+    content {
+      name     = var.cert_name
+      data     = var.cert_file
+      password = var.cert_password
+    }
+
+  }
+
   lifecycle {
     # K8S will be changing all of these settings so we ignore them.
     # We really only needed this resource to assign a known public IP.
     ignore_changes = [
-      ssl_certificate,
+      # ssl_certificate,
```

Comment:
I'm concerned that this will lead to a lot of noise in clusters that aren't using BYOC.  Can you make this entry dynamic based on if byoc is enabled or not?

Reply:
I tried. However, the lifecycle block has no support for dynamic nor accepts configuration from local variables.
I will revert this change. The main point here is that, if the customer needs to update their certificate, they will need to do it manually It's not a big deal, and it's better than every time the LetsEncrpypt update the cert, we will have a huge drift across our fleet.